apiml-security-common/test-programs/CheckPerm.java

```diff
@@ -0,0 +1,65 @@
+import com.ibm.os390.security.PlatformAccessControl;
+import com.ibm.os390.security.PlatformReturned;
+
+public class CheckPerm {
+    static int failed = 0;
+
+    static PlatformReturned checkPermission(String userid, String resourceClass, String resourceName, int accessLevel,
+            int expectedErrno, int expectedErrno2, int secondExpectedErrno, int secondExpectedErrno2) {
+        System.out.print(String.format("PlatformAccessControl.checkPermission(%s, %s, %s, %d) = ", userid,
+                resourceClass, resourceName, accessLevel));
+        PlatformReturned r = PlatformAccessControl.checkPermission(userid,
+                resourceClass, resourceName, accessLevel);
+        System.out.println(returnedToString(r));
+        if (expectedErrno == 0) {
+            if (r == null) {
+                System.out.println("OK - null expected");
+            } else {
+                System.out.println("ERROR - null expected but returned: " + returnedToString(r));
+                failed++;
+            }
+        } else {
+            if (r == null) {
+                System.out.println("ERROR - unexpected errno and errno2 - expected null");
+                failed++;
+            } else if (((r.errno == expectedErrno) && (r.errno2 == expectedErrno2)) || ((r.errno == secondExpectedErrno) && (r.errno2 == secondExpectedErrno2))) {
+                System.out.println("OK - expected errno and errno2 returned");
+            } else {
+                System.out.println(String.format("ERROR - unexpected errno and errno2 returned. Expected %d and 0x%04x",
+                        expectedErrno, expectedErrno2));
+                failed++;
+            }
+        }
```

Comment:
```suggestion
            if (r == null) {
                System.out.println("ERROR - unexpected errno and errno2 - expected null");
                failed++;
            } else if (((r.errno == expectedErrno) && (r.errno2 == expectedErrno2)) || ((r.errno == secondExpectedErrno) && (r.errno2 == secondExpectedErrno2))) {
                System.out.println("OK - expected errno and errno2 returned");
		failed = (expectedErrno == 0);
            } else {
                System.out.println(String.format("ERROR - unexpected errno and errno2 returned. Expected %d and 0x%04x",
                        expectedErrno, expectedErrno2));
                failed++;
            }
```

Reply:
It is shorted but it does not the same. I will keep the original that has more detailed messages.